crates/miden-lib/asm/kernels/transaction/lib/memory.masm

```diff
@@ -1755,6 +1756,24 @@ export.set_output_note_assets_hash
     mem_storew
 end
 
+#! Returns the assets commitment for the output note located at the specified memory address.
+#!
+#! TODO: check this statement
+#! Notice that this hash is available only after `tx::compute_output_notes_commitment` procedure has
+#! been called, otherwise this procedure will return an empty word.
```

Comment:
I would modify the comment to say that this returns the commitment to the note's assets as of the last time `tx::compute_output_notes_commitment` was called.

Also, maybe we should split the part that updates the asset commitment out of the `tx::compute_output_notes_commitment` into a separate procedure - something like `tx::compute_output_note_asset_commitment` (if this make sense, of course).